### DISCUSSION TITLE
Check effectiveness of 5.0.0 fix to memory leak in NamespaceManager

### DISCUSSION THREAD

**Discussion by joernhees** [RDFLib/rdflib#1572] [TO-DO list]
in its current form `NamespaceManager` has a `__cache = {}` attrib:
https://github.com/RDFLib/rdflib/blob/master/rdflib/namespace.py#L286

while there is a `reset()` i am living proof that this can cause days of debugging to find a memory leak... turns out `NamespaceManager` held refs to all URIs it ever saw... great.

If i find time and no one objects, i'd add `cachetools` as dependency and use a `LRUCache` and make its size configurable....

**Comment by nicholascar** [comment]
This needs testing for 6.0.0 to see if the namespace tidy-up functions added in 5.0.0 have changed this scenario.
